src/test/java/org/opensearch/integTest/RAGToolIT.java

```diff
@@ -0,0 +1,327 @@
+            + "          \"text\": \"embedding\"\n"
+            + "        }\n"
+            + "      }\n"
+            + "    }\n"
+            + "  ]\n"
+            + "}";
+        createIngestPipelineWithConfiguration("test-embedding-model", pipelineConfig);
+
+        String indexMapping = "{\n"
+            + "  \"mappings\": {\n"
+            + "    \"properties\": {\n"
+            + "      \"text\": {\n"
+            + "        \"type\": \"text\"\n"
+            + "      },\n"
+            + "      \"embedding\": {\n"
+            + "        \"type\": \"knn_vector\",\n"
+            + "        \"dimension\": 384,\n"
+            + "        \"method\": {\n"
+            + "          \"name\": \"hnsw\",\n"
+            + "          \"space_type\": \"l2\",\n"
+            + "          \"engine\": \"lucene\",\n"
+            + "          \"parameters\": {\n"
+            + "            \"ef_construction\": 128,\n"
+            + "            \"m\": 24\n"
+            + "          }\n"
+            + "        }\n"
+            + "      }\n"
+            + "    }\n"
+            + "  },\n"
+            + "  \"settings\": {\n"
+            + "    \"index\": {\n"
+            + "      \"knn.space_type\": \"cosinesimil\",\n"
+            + "      \"default_pipeline\": \"test-embedding-model\",\n"
+            + "      \"knn\": \"true\"\n"
+            + "    }\n"
+            + "  }\n"
+            + "}";
+
+        createIndexWithConfiguration(TEST_NEURAL_INDEX_NAME, indexMapping);
+
+        addDocToIndex(TEST_NEURAL_INDEX_NAME, "0", List.of("text"), List.of("hello world"));
+
+        addDocToIndex(TEST_NEURAL_INDEX_NAME, "1", List.of("text"), List.of("a b"));
+    }
+
+    @Before
+    @SneakyThrows
+    public void setUp() {
+        super.setUp();
+        prepareModel();
+        prepareIndex();
+        String registerAgentWithNeuralQueryRequestBodyFile = Files
+            .readString(
+                Path
+                    .of(
+                        this
+                            .getClass()
+                            .getClassLoader()
+                            .getResource(
+                                "org/opensearch/agent/tools/register_flow_agent_of_ragtool_with_neural_query_type_request_body.json"
+                            )
+                            .toURI()
+                    )
+            );
+        registerAgentWithNeuralQueryRequestBody = registerAgentWithNeuralQueryRequestBodyFile
+            .replace("<MODEL_ID>", textEmbeddingModelId)
+            .replace("<INDEX_NAME>", TEST_NEURAL_INDEX_NAME);
+
+        registerAgentWithNeuralSparseQueryRequestBody = registerAgentWithNeuralQueryRequestBodyFile
+            .replace("<MODEL_ID>", sparseEncodingModelId)
+            .replace("<INDEX_NAME>", TEST_NEURAL_SPARSE_INDEX_NAME)
+            .replace("\"query_type\": \"neural\"", "\"query_type\": \"neural_sparse\"");
+    }
+
+    @After
+    @SneakyThrows
+    public void tearDown() {
```

Comment:
should also delete model here

Reply:
ref: https://github.com/opensearch-project/skills/blob/e9632040e47b24963a86b38a68d723d530033eeb/src/test/java/org/opensearch/integTest/NeuralSparseSearchToolIT.java#L91

Reply:
added, thanks!